pandas/tests/util/test_util.py

```diff
@@ -417,6 +417,29 @@ def test_numpy_errstate_is_default():
     assert np.geterr() == expected
 
 
+@td.skip_if_windows
+def test_can_set_locale_invalid_set():
+    # Setting an invalid locale should return False
+    assert tm.can_set_locale('non-existent_locale') is False
+
+
+@td.skip_if_windows
+def test_can_set_locale_invalid_get(monkeypatch):
+    # In some cases, an invalid locale can be set,
+    # but a subsequent getlocale() raises a ValueError
+    # See GH 22129
+
+    def mocksetlocale(category, locale):
+        return locale
+
+    def mockgetlocale():
+        raise ValueError()
+
+    monkeypatch.setattr(locale, 'setlocale', mocksetlocale)
```

Comment:
Can you use the context manager for patching instead? From the monkey patch docs it is the preferred watching of patching stdlib